### PR TITLE
Add calico e2e test to aws clusterclass suite

### LINKED ISSUE #2460: TC11: AWS Cluster - Ability to create CC guest cluster on AWS with calico CNI and verify it is successful

### PR DESCRIPTION
**Pull request by codegold79**
Signed-off-by: F. Gold <fgold@vmware.com>, Leo Li  <lhaoran@vmware.com>

### What this PR does / why we need it
- Add Calico CNI E2E tests to AWS_CC suite
### Which issue(s) this PR fixes

Fixes #2460

### Describe testing done for PR
- E2E tests passed locally

### Release note

```release-note
Add Calico CNI E2E tests to AWS_CC suite
```

### CODE REVIEW

File: pkg/v1/tkg/test/tkgctl/shared/package.go [hunk 1]
```diff
@@ -267,7 +267,7 @@ func verifyClusterBootstrap(ctx context.Context, c client.Client, clusterBootstr
 		expectedClusterBootstrap.Spec.CNI.ValuesFrom.ProviderRef = &corev1.TypedLocalObjectReference{
 			APIGroup: &apiGroup,
 			Kind:     "CalicoConfig",
-			Name:     GeneratePackageSecretName(clusterBootstrap.Name, "calico"),
```
Comment (maralavi) [inline pkg/v1/tkg/test/tkgctl/shared/package.go:270]: for clusters made with custom CB, the name of the config object is the same as the CB. But for those CB created from CBT, the above name is used. So both these names can be valid for different scenarios. 
Now why this change is needed?

File: pkg/v1/tkg/test/tkgctl/shared/e2e_common.go [hunk 2]
```diff
@@ -216,7 +216,7 @@ func E2ECommonSpec(ctx context.Context, inputGetter func() E2ECommonSpecInput) {
 				By(fmt.Sprintf("Verify workload cluster %q resources have been deleted", clusterName))
 				Eventually(func() bool {
 					return clusterResourcesDeleted(ctx, mngClient, clusterResources)
-				}, waitTimeout, pollingInterval).Should(BeTrue())
+				}, 2*waitTimeout, pollingInterval).Should(BeTrue())
```
Comment (maralavi) [inline pkg/v1/tkg/test/tkgctl/shared/e2e_common.go:219]: Why increasing the timeout, and what guarantees this is enough? Why current timeout used to work before?

File: pkg/v1/tkg/test/tkgctl/shared/package.go (hunk 1 above)
Comment (HL-EverGreen) [reply pkg/v1/tkg/test/tkgctl/shared/package.go:270]: Thanks for the review! As Shyaam mentioned in the channel, this change is in the path of Calico CNI. And since Calico is the case of custom CB, I think it is safe to change here. The CB created from CBT won't go in this path as the CNI won't be Calico.

File: pkg/v1/tkg/test/tkgctl/shared/e2e_common.go (hunk 2 above)
Comment (HL-EverGreen) [reply pkg/v1/tkg/test/tkgctl/shared/e2e_common.go:219]: Hi @maralavi, the old timeout didn't work for Calico tests before. The test was flaky due to timeout in resource deletion. After increasing the timeout I didn't see that issue for several runs. So I think it helps mitigate the flaky scenario.

File: pkg/v1/tkg/test/tkgctl/shared/package.go (hunk 1 above)
Comment (maralavi) [reply pkg/v1/tkg/test/tkgctl/shared/package.go:270]: Ok, so just noticed ```strings.Contains(clusterBootstrap.Spec.CNI.RefName, "calico")``` check, and calico always use custom CB. We should be good then. Thanks for the clarification.

**Review by maralavi — APPROVED** [review]
